web/src/connectors/task-annotator-connector/utils.ts

```diff
@@ -0,0 +1,129 @@
+import { CategoryDataAttrType } from '../../api/typings';
+import { PageSize } from '../../shared/components/document-pages/document-pages';
+
+interface Token {
+    id: number;
+    text: string;
+    x: number;
+    y: number;
+    width: number;
+    height: number;
+}
+
+interface Data {
+    tokens: Token[];
+    dataAttributes: CategoryDataAttrType[];
+}
+
+interface PageInfoObj {
+    id: number;
+    type: string;
+    bbox: number[];
+    category: string;
+    data: Data;
+    children: number[];
+    text: string;
+}
+
+interface Page {
+    page_num: number;
+    size: PageSize;
+    objs: PageInfoObj[];
+    revision: string;
+    user_id: string;
+    pipeline: number;
+    date: string;
+    is_validated: boolean;
+    categories: string[];
+}
+
+export interface LatestRevisionObj {
+    [page_num: string]: Page[];
+}
+
+export interface RevisionObjByUser {
+    [user_id: string]: {
+        [page_num: number]: {
+            size: PageSize;
+            objs: PageInfoObj[];
+        };
+    };
+}
+
+export interface UserRevision {
+    user_id: string;
+    page_num: number;
+    size: PageSize;
+    objs: PageInfoObj[];
+    revision: string;
+    isValidated: boolean;
+    date: string;
+    pipeline: number;
+    categories: string[];
+}
+
+export function convertToRevisionByUserArray(sourceObj: LatestRevisionObj): UserRevision[] {
+    const userRevisions: UserRevision[] = [];
+
+    for (const page_num in sourceObj) {
```

Comment:
Maybe we can just get Object.values(sourceObj) and iterate it? Or maybe even Object.values(sourceObj).flat()

Reply:
fixed https://github.com/epam/badgerdoc/pull/584/commits/c3070a3ede64aecc9854ce329ac318ca85fff31e